src/components/Dialog/DialogsManager.ts

```diff
@@ -0,0 +1,178 @@
+    DialogsManagerEvent['type'],
+    DialogsManagerEventHandler[]
+  > = { openCountChange: [] };
+
+  constructor({ id }: DialogInitOptions = {}) {
+    this.id = id ?? new Date().getTime().toString();
+  }
+
+  getOrCreate({ id, isOpen = false }: GetOrCreateParams) {
+    let dialog = this.dialogs[id];
+    if (!dialog) {
+      dialog = {
+        close: () => {
+          this.close(id);
+        },
+        id,
+        isOpen,
+        open: () => {
+          this.open({ id });
+        },
+        remove: () => {
+          this.remove(id);
+        },
+        toggle: () => {
+          this.toggleOpen({ id });
+        },
+        toggleSingle: () => {
+          this.toggleOpenSingle({ id });
+        },
+      };
+      this.dialogs[id] = dialog;
+    }
+    return dialog;
+  }
+
+  on(
+    eventType: DialogEvent['type'] | DialogsManagerEvent['type'],
+    { id, listener }: { listener: DialogEventHandler | DialogsManagerEventHandler; id?: DialogId },
+  ) {
+    if (dialogsManagerEvents.includes(eventType as DialogsManagerEvent['type'])) {
+      this.dialogsManagerEventListeners[eventType as DialogsManagerEvent['type']].push(
+        listener as DialogsManagerEventHandler,
+      );
+      return () => {
+        this.off(eventType, { listener });
+      };
+    }
+    if (!id) return noop;
+
+    if (!this.dialogEventListeners[id]) {
+      this.dialogEventListeners[id] = { close: [], open: [] };
+    }
+    this.dialogEventListeners[id][eventType] = [
+      ...(this.dialogEventListeners[id][eventType] ?? []),
+      listener as DialogEventHandler,
+    ];
+    return () => {
+      this.off(eventType, { id, listener });
+    };
+  }
+
+  off(
+    eventType: DialogEvent['type'] | DialogsManagerEvent['type'],
+    { id, listener }: { listener: DialogEventHandler | DialogsManagerEventHandler; id?: DialogId },
+  ) {
+    if (dialogsManagerEvents.includes(eventType as DialogsManagerEvent['type'])) {
+      const eventListeners = this.dialogsManagerEventListeners[
+        eventType as DialogsManagerEvent['type']
+      ];
+      eventListeners?.filter((l) => l !== listener);
+      return;
+    }
+
+    if (!id) return;
+
+    const eventListeners = this.dialogEventListeners[id]?.[eventType];
+    if (!eventListeners) return;
+    this.dialogEventListeners[id][eventType] = eventListeners.filter((l) => l !== listener);
+  }
+
+  open(params: GetOrCreateParams, single?: boolean) {
+    const dialog = this.getOrCreate(params);
+    if (dialog.isOpen) return;
+    if (single) {
+      this.closeAll();
+    }
+    this.dialogs[params.id].isOpen = true;
+    this.openDialogCount++;
+    this.dialogsManagerEventListeners.openCountChange.forEach((listener) => listener(this));
+    this.dialogEventListeners[params.id].open?.forEach((listener) => listener(dialog));
+  }
+
+  close(id: DialogId) {
+    const dialog = this.dialogs[id];
+    if (!dialog?.isOpen) return;
+    dialog.isOpen = false;
+    this.openDialogCount--;
```

Comment:
Once this class is converted to using `StateStore`, this property could be a cached getter deriving the value from the list of open dialogs.

Reply:
I want to avoid array two arrays of open and closed dialogs and iterate each time to find it just to have a getter prop.  Not sure, if I understood what you propose.

Reply:
I will remove openDialogCount and introduce the subscription selector that counts the number of open dialogs and that way the getter will not be necessary.

Reply:
I think for the amount of active dialogs the subscription with selector should suffice. 👍

Reply:
Plays nice with tests